modules/service/src/main/scala/lucuma/odb/graphql/mapping/MutationMapping.scala

```diff
@@ -208,12 +247,12 @@ trait MutationMapping[F[_]: MonadCancelThrow] extends Predicates[F] {
       val selectObservations: F[Result[(List[Observation.Id], Query)]] =
         idSelect.traverse { which =>
           observationService.use { svc =>
-            svc
-              .selectObservations(which)
-              .fproduct {
-                case Nil => Limit(0, child)
-                case ids => Filter(Predicates.observation.id.in(ids), child)
```

Comment:
This dance isn't necessary anymore. `In(Nil)` now correctly evaluates to `false`.

Reply:
Cool.